src/engine/effects/message.h

```diff
@@ -40,17 +40,9 @@ struct EffectsRequest {
               value(0.0) {
         pTargetChain = nullptr;
         pTargetEffect = nullptr;
-#define CLEAR_STRUCT(x) memset(&x, 0, sizeof(x));
-        CLEAR_STRUCT(AddEffectChain);
-        CLEAR_STRUCT(RemoveEffectChain);
-        CLEAR_STRUCT(EnableInputChannelForChain);
-        CLEAR_STRUCT(DisableInputChannelForChain);
-        CLEAR_STRUCT(AddEffectToChain);
-        CLEAR_STRUCT(RemoveEffectFromChain);
-        CLEAR_STRUCT(SetEffectChainParameters);
-        CLEAR_STRUCT(SetEffectParameters);
-        CLEAR_STRUCT(SetParameterParameters);
-#undef CLEAR_STRUCT
+        // zero out the struct with the largest size to ensure the entire union memory is set to
+        // zero.
+        memset(&SetEffectChainParameters, 0, sizeof(SetEffectChainParameters));
```

Comment:
Can we name the anonymous union here and then clear the memory by sizeof(union Payload)?
All other ideas I have in mind are breaking during refectoring. 

However, I actually think that a zeroed memory is in the same way dangerous as a ununutalized memory.
We must not use it in both cases. It looks like the rest of the control structures take already care of it. So if zeroing has an effect, it will only obfusticate an issue we have anyway. 

Thats why I would prefere to not initialize the memory. It makes more sense to asserrt, the validity of the control structures and introduce constructors, that take care of them in one call and cannot fail.